File: historyscan-back/src/main/java/io/sch/historyscan/infrastructure/features/analysis/clocrevisions/FileInfoDTO.java
package io.sch.historyscan.infrastructure.features.analysis.clocrevisions;

import com.fasterxml.jackson.annotation.JsonCreator;
import org.springframework.hateoas.RepresentationModel;

import java.util.Objects;

public class FileInfoDTO extends RepresentationModel<FileInfoDTO> {
    private final String name;
    private final String path;

    @JsonCreator
    public FileInfoDTO(String name, String path) {
        this.name = name;
        this.path = path;
    }

    public String getName() {
        return name;
    }

    public String getPath() {
        return path;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        FileInfoDTO that = (FileInfoDTO) o;
        return Objects.equals(name, that.name) && Objects.equals(path, that.path);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), name, path);
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/infrastructure/features/analysis/dto/FileInfoDTO.kt
package io.sch.historyscan.infrastructure.features.analysis.dto

import com.fasterxml.jackson.annotation.JsonCreator
import org.springframework.hateoas.RepresentationModel
import java.util.*

class FileInfoDTO @JsonCreator constructor(val name: String, val path: String) : RepresentationModel<FileInfoDTO?>() {
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        if (!super.equals(o)) return false
        val that = o as FileInfoDTO
        return name == that.name && path == that.path
    }

    override fun hashCode(): Int {
        return Objects.hash(super.hashCode(), name, path)
    }
}
